src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -116,18 +116,19 @@ void TimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
     VerifyOrReturn(delegate != nullptr, ChipLogError(Zcl, "Delegate is null. Unable to handle timer expired"));
 
     delegate->ClearPendingPresetList();
-    gThermostatAttrAccess.SetPresetsEditable(endpoint, false);
+    gThermostatAttrAccess.SetAtomicWrite(endpoint, false);
+    gThermostatAttrAccess.SetAtomicWriteScopedNodeId(endpoint, ScopedNodeId());
 }
 
 /**
- * @brief Schedules a timer for the given timeout in seconds.
+ * @brief Schedules a timer for the given timeout in milliseconds.
  *
  * @param[in] endpoint The endpoint to use.
- * @param[in] timeoutSeconds The timeout in seconds.
+ * @param[in] timeoutMilliseconds The timeout in milliseconds.
  */
-void ScheduleTimer(EndpointId endpoint, uint16_t timeoutSeconds)
+void ScheduleTimer(EndpointId endpoint, uint16_t timeoutMilliseconds)
```

Comment:
How about we just have this take a System::Clock::Milliseconds16 argument to start with?